### PR TITLE
optional timeout parameter for the dd exec

### PR DESCRIPTION
**Pull request by BIAndrews**
rake spec successful, new test included. This is needed in AWS EC2 instances with more memory than disk I/O bandwidth. The exec can run into a timeout in some cases. 

~~~
swap_file::files
  default parameters
    should compile into a catalogue without dependency cycles
    should contain Exec[Create swap file /mnt/swap.1] with command => "/bin/dd if=/dev/zero of=/mnt/swap.1 bs=1M count=1073" and creates => "/mnt/swap.1"
    should contain File[/mnt/swap.1] with owner => "root", group => "root", mode => "0600" and require => "Exec[Create swap file /mnt/swap.1]"
    should contain Exec[Attach swap file /mnt/swap.1] with command => "/sbin/mkswap /mnt/swap.1 && /sbin/swapon /mnt/swap.1", require => "File[/mnt/swap.1]" and unless => "/sbin/swapon -s | grep /mnt/swap.1"
    should contain Mount[/mnt/swap.1] with require => "Exec[Attach swap file /mnt/swap.1]"
  custom swapfilesize parameter
    should compile into a catalogue without dependency cycles
    should contain Exec[Create swap file /mnt/swap.1] with command => "/bin/dd if=/dev/zero of=/mnt/swap.1 bs=1M count=4402" and creates => "/mnt/swap.1"
  custom swapfilesize parameter with timeout
    should compile into a catalogue without dependency cycles
    should contain Exec[Create swap file /mnt/swap.2] with command => "/bin/dd if=/dev/zero of=/mnt/swap.2 bs=1M count=4402" and creates => "/mnt/swap.2"

Finished in 1.37 seconds (files took 1.06 seconds to load)
9 examples, 0 failures
~~~

### CODE REVIEW

File: spec/defines/files_spec.rb [hunk 1]
```diff
@@ -65,4 +65,25 @@
...
+        :timeout => 900,
+      }
+    end
+    it do
+      is_expected.to compile.with_all_deps
+    end
+    it do
+      is_expected.to contain_exec('Create swap file /mnt/swap.2').
+      with({"command"=>"/bin/dd if=/dev/zero of=/mnt/swap.2 bs=1M count=4402",
+       "creates"=>"/mnt/swap.2"})
```
Comment (petems) [inline spec/defines/files_spec.rb]: Can you add `"timeout" => 900` to this? Will actually test the exec has the right timeout.

**Comment by petems** [comment]
@BIAndrews Very nice about the fallocate btw, I mean to do something about that was thinking about `truncate`, but according to this fallocate is the best so that's great :+1:  http://hasis053341.blogspot.co.uk/2014/08/using-fallocate-instead-of-dd-for.html

**Comment by BIAndrews** [comment]
Ok after this PR I'll send another with the fallocate cmd option. I have it done already and the spec setup with a test all ready to go for you.

**Comment by petems** [comment]
Rebased into https://github.com/petems/puppet-swap_file/pull/27

**Pull request closed without merging**